### PR TITLE
Deduplicate tag values in metrics actuator endpoint

### PR DESCRIPTION
**Pull request by jkschneider**
cc / @joshiste

### CODE REVIEW

File: spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/MetricsEndpoint.java [hunk 1]
```diff
@@ -16,14 +16,7 @@
...
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
-import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (snicoll) [inline spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/MetricsEndpoint.java]: Can you please rework the PR so that it doesn't change formatting or anything that's not directly related to the change?

File: spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/metrics/MetricsEndpointTests.java [hunk 2]
```diff
@@ -77,7 +77,6 @@ public void metricValuesAreTheSumOfAllTimeSeriesMatchingTags() {
 		this.registry.counter("cache", "result", "hit", "host", "1").increment(2);
 		this.registry.counter("cache", "result", "miss", "host", "1").increment(2);
 		this.registry.counter("cache", "result", "hit", "host", "2").increment(2);
-		MockClock.clock(this.registry).add(SimpleConfig.DEFAULT_STEP);
```
Comment (snicoll) [inline spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/metrics/MetricsEndpointTests.java]: Is that related?

File: spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/metrics/MetricsEndpointTests.java [hunk 3]
```diff
@@ -88,10 +87,22 @@ public void metricValuesAreTheSumOfAllTimeSeriesMatchingTags() {
...
+				.stream()
+				.filter(t -> t.getTag().equals("region"))
+				.flatMap(t -> t.getValues().stream()))
+				.containsExactly("east");
+	}
+
 	@Test
 	public void metricWithSpaceInTagValue() {
 		this.registry.counter("counter", "key", "a space").increment(2);
-		MockClock.clock(this.registry).add(SimpleConfig.DEFAULT_STEP);
```
Comment (snicoll) [inline spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/metrics/MetricsEndpointTests.java]: Is that related?

Comment (jkschneider) [reply spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/metrics/MetricsEndpointTests.java]: This line is unnecessary as of rc5

File: spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/metrics/MetricsEndpointTests.java (hunk 2 above)
Comment (jkschneider) [reply spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/metrics/MetricsEndpointTests.java]: This line is unnecessary as of rc5

File: spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/metrics/MetricsEndpointTests.java (hunk 3 above)
Comment (snicoll) [reply spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/metrics/MetricsEndpointTests.java]: Thanks but as it is not related to the purpose of this PR, I've polished this separately in https://github.com/spring-projects/spring-boot/commit/1962cbb6234dd9029bf6a4bb6374d793ba5de959
